lib/platform/gitea/index.ts

```diff
@@ -0,0 +1,361 @@
+import URL from 'url';
+import is from '@sindresorhus/is';
+
+import api from './gl-got-wrapper';
+import * as hostRules from '../../util/host-rules';
+import GitStorage from '../git/storage';
+import { PlatformConfig } from '../common';
+
+let config: {
+  storage: GitStorage;
+  repository: string;
+  localDir: string;
+  defaultBranch: string;
+  baseBranch: string;
+  email: string;
+  prList: any[];
+  issueList: any[];
+  token: string;
+} = {} as any;
+
+const defaults = {
+  hostType: 'gitea',
+  endpoint: 'https://gitea.com/api/v1/',
+};
+
+export async function initPlatform({
+  endpoint,
+  token,
+}: {
+  endpoint: string;
+  token: string;
+}) {
+  if (!token) {
+    throw new Error('Init: You must configure a Gitea personal access token');
+  }
+  const res = {} as any;
+  if (endpoint) {
+    res.endpoint = endpoint.replace(/\/?$/, '/'); // always add a trailing slash
+    api.setBaseUrl(res.endpoint);
+    defaults.endpoint = res.endpoint;
+  } else {
+    res.endpoint = defaults.endpoint;
+    logger.info('Using default Gitea endpoint: ' + res.endpoint);
+  }
+  try {
+    res.gitAuthor = (await api.get(`user?token=${token}`)).body.email;
```

Comment:
Let's go with the header-based approach for tokens:

![image](https://user-images.githubusercontent.com/6311784/67096982-8094f400-f1b9-11e9-9f94-4e6cbeacbcef.png)